pkg/distributor/distributor.go

```diff
@@ -401,6 +403,20 @@ func (d *Distributor) sendRequests(ctx context.Context, req *distributormodel.Pu
 
 	// Next we split profiles by labels.
 	profileSeries := extractSampleSeries(req)
+
+	// Apply relabeling rules to the series.
+	relabelingRules := d.limits.IngestionRelabelingRules(tenantID)
+	builder := phlaremodel.NewLabelsBuilder(phlaremodel.EmptyLabels())
+	for _, series := range profileSeries {
```

Comment:
I think there's a problem with relabeling after `extractSampleSeries`: say, we have two series formed by pprof labels:
```
{foo=bar, baz=qux}
{foo=bar, baz=zoo}
```

Then, we drop the `baz` label:
```
{foo=bar}
{foo=bar}
```

The problem is that both profiles have the same timestamp (and even profile ID, IIRC), which means that one will "replace" another at ingestion (or will be deduplicated at reads). Probably same applies to the explicit series labels.

A simple solution for this would be merging such "matching" profiles, but merges are relatively expensive. It would be great if we could do all the relabeling stuff during the series extraction from pprof samples.

Also, not sure if it's feasible (rules must be idempotent) but I'd consider first applying them on the explicit series labels